options.go

```diff
@@ -120,8 +121,29 @@ func HashRingConfig(c *hashring.Configuration) Option {
 // automatically.
 func Logger(l log.Logger) Option {
 	return func(r *Ringpop) error {
-		r.logger = l
-		r.log = l
+		if r.moduleLogger == nil {
+			r.moduleLogger = rutil.NewModuleLogger(l)
+		} else {
+			r.moduleLogger.SetLogger(rutil.RootModule, l)
+		}
+		return nil
+	}
+}
+
+// ModuleLevel is used to configure different log levels per module. Messages
+// with a level less than minLevel are silenced. By default, an unconfigured
+// module outputs all messages.
+func ModuleLevel(name string, minLevel rutil.Level) Option {
```

Comment:
Maybe we should have another chat to go through it. You'll need to explain to me what a "per module" logger is, and why we would want to pass one at runtime.